src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala

```diff
@@ -295,20 +155,37 @@ case class GenericRecordReader(callFields: Set[String]) extends HtsjdkRecordRead
                   case e: Exception =>
                     fatal(
                       s"""variant $v: Genotype field ${ f.name }:
-                 |  unable to convert $x (of class ${ x.getClass.getCanonicalName }) to ${ f.typ }:
-                 |  caught $e""".stripMargin)
+                         |  unable to convert $x (of class ${ x.getClass.getCanonicalName }) to ${ f.typ }:
+                         |  caught $e""".stripMargin)
                 }
               }
             }
 
           try {
-            HtsjdkRecordReader.cast(a, f.typ)
+            var r = HtsjdkRecordReader.cast(a, f.typ)
+
+            // handle diploid
```

Comment:
Not new, I had to port this from GentypeRecordReader to get testHaploid to pass.  I think once all of this is done, we'll be able to generalize `Call` easily to support multi-ploidy and phasing.